### ISSUE TITLE
[CAL-4487] Update GitHub workflows to install playwright in parallel

### ISSUE THREAD

**Issue by keithwillcode** [calcom/cal.com#17025]
### Is your proposal related to a problem?

Inside of our "Yarn install & cache" job, it's 1) installing playwright even if the E2E tests aren't set to run (based on the ready-for-e2e label) and 2) blocking other jobs that don't need this to be installed.

### Describe the solution you'd like

Move the playwright installation to a separate job that runs before the E2E test suite, unblocking other jobs like the builds and unit tests that don't need playwright.

This will offload around 40s every time new PR Update workflows are started (cached).

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4487](https://linear.app/calcom/issue/CAL-4487/update-github-workflows-to-install-playwright-in-parallel)</sub>

**Comment by Aditya-A-G** [comment]
hey, I would like to work on this issue
